linera-core/src/updater.rs

```diff
@@ -227,10 +223,37 @@ where
             Err(original_err @ NodeError::BlobsNotFound(blob_ids)) => {
                 self.remote_node
                     .check_blobs_not_found(&certificate, blob_ids)?;
+                // The certificate is confirmed, so the blobs must be in storage.
+                let maybe_blobs = self.local_node.read_blobs(blob_ids).await?;
+                let blobs = maybe_blobs.ok_or_else(|| original_err.clone())?;
+                self.remote_node
+                    .handle_certificate(certificate, blobs, delivery)
+                    .await
+            }
+            _ => result,
+        }?)
+    }
 
+    async fn send_validated_certificate(
+        &mut self,
+        certificate: GenericCertificate<ValidatedBlock>,
+        delivery: CrossChainMessageDelivery,
+    ) -> Result<Box<ChainInfo>, ChainClientError> {
+        let result = self
+            .remote_node
+            .handle_optimized_certificate(&certificate, delivery)
+            .await;
+
+        Ok(match &result {
+            Err(original_err @ NodeError::BlobsNotFound(blob_ids)) => {
+                self.remote_node
+                    .check_blobs_not_found(&certificate, blob_ids)?;
+                let chain_id = certificate.inner().executed_block().block.chain_id;
+                // The certificate is for a validated block, i.e. for our locked block.
+                // Take the missing blobs from our local chain manager.
                 let blobs = self
                     .local_node
-                    .find_missing_blobs(blob_ids.clone(), certificate.inner().chain_id())
+                    .find_locked_blobs(blob_ids, chain_id)
```

Comment:
Since `find_locked_blobs` seems to be returning `None` if any of the `blob_ids` is not found maybe we can rename it to `get_locked_blobs`?

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3006/commits/237cc7b75d37e7facb210cd7f08d423555aa734e.